impeller/renderer/backend/vulkan/allocator_vk.cc

```diff
@@ -90,12 +127,67 @@ AllocatorVK::AllocatorVK(std::weak_ptr<Context> context,
     VALIDATION_LOG << "Could not create memory allocator";
     return;
   }
+  if (!CreateBufferPool(allocator, &raster_buffer_pool_)) {
+    return;
+  }
+  if (!CreateBufferPool(allocator, &image_upload_buffer_pool_)) {
+    return;
+  }
+
+  {
+    vk::ImageCreateInfo image_info;
+    image_info.flags = {};
+    image_info.imageType = vk::ImageType::e2D;
+    image_info.format = vk::Format::eR8G8B8A8Unorm;
+    image_info.extent = VkExtent3D{1u, 1u, 1u};
+    image_info.samples = vk::SampleCountFlagBits::e1;
+    image_info.mipLevels = 1u;
+    image_info.arrayLayers = 1u;
+    image_info.tiling = vk::ImageTiling::eOptimal;
+    image_info.initialLayout = vk::ImageLayout::eUndefined;
+    image_info.usage =
+        vk::ImageUsageFlagBits::eSampled | vk::ImageUsageFlagBits::eTransferDst;
+    image_info.sharingMode = vk::SharingMode::eExclusive;
+
+    auto create_info_native =
+        static_cast<vk::ImageCreateInfo::NativeType>(image_info);
+
+    VmaAllocationCreateInfo sampleAllocCreateInfo = {};
+    sampleAllocCreateInfo.usage = VMA_MEMORY_USAGE_AUTO;
+    sampleAllocCreateInfo.preferredFlags = VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT;
+
+    uint32_t memTypeIndex;
+    VkResult res = vmaFindMemoryTypeIndexForImageInfo(
+        allocator, &create_info_native, &sampleAllocCreateInfo, &memTypeIndex);
+
+    // Create a pool that can have at most 2 blocks, 128 MiB each.
+    VmaPoolCreateInfo poolCreateInfo = {};
+    poolCreateInfo.memoryTypeIndex = memTypeIndex;
+    poolCreateInfo.blockSize = 128ull * 1024 * 1024;
```

Comment:
As mentioned in https://github.com/flutter/engine/pull/43313#discussion_r1246168700, let's put the tunables in a separate file so everything is available at a glance?